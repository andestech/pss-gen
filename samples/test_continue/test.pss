component pss_top {
    list<int> suffix;
    list<int> l;
    exec init_down {
        int i = 100;
        repeat (10) {
            suffix.push_back(i);
            i = i + 1;
        }
    }
    exec init_up {
        int i = 0;
        repeat (10) {
           i = i + 1;
           if (i % 2 == 0) continue;
           l.push_back(i);
        }
        while (i < 20) {
           i = i + 1;
           if (i % 2 == 0) continue;
           l.push_back(i);
        }
        repeat {
           i = i + 1;
           if (i % 2 == 0) continue;
           l.push_back(i);
        } while (i < 30);
        foreach (s : suffix) {
            i = i + 1;
            if (i % 2 == 0) continue;
            l.push_back (s);
        }
    }
    action print {
        rand int out;
        exec body ASM = """out = {{out}}""";
    }
    action root_a {
        activity {
            foreach (v : pss_top.l) {
                do print with { out == v; };
            }
        }
    }
}

