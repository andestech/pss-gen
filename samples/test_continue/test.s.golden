out = 1
out = 3
out = 5
out = 7
out = 9
out = 11
out = 13
out = 15
out = 17
out = 19
out = 21
out = 23
out = 25
out = 27
out = 29
